src/main/java/com/google/api/codegen/DiscoveryContext.java

```diff
@@ -84,6 +84,108 @@ public String getSimpleName(String name) {
     return name.substring(name.lastIndexOf('.') + 1);
   }
 
+  public String getSampleVarName(String typeName) {
+    return upperCamelToLowerCamel(getSimpleName(typeName));
```

Comment:
The return value is correctly formatted for all languages? If this is meant to be overriden by each language, please comment that.